fvtest/compilerunittest/p/BinaryEncoder.cpp

```diff
@@ -0,0 +1,2856 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "../CodeGenTest.hpp"
+#include "codegen/GenerateInstructions.hpp"
+#include "il/LabelSymbol.hpp"
+
+#include <iostream>
+#include <stdexcept>
+
+class BinaryInstruction {
+public:
+    std::vector<uint32_t> _words;
+
+    BinaryInstruction() {}
+
+    BinaryInstruction(uint32_t instr) {
```

Comment:
No, this constructor is deliberately implicit. This is done to reduce the amount of boilerplate when adding tests. Since the vast majority of our instructions encode to 4 bytes (only pseudoinstructions and certain expanding memory instructions currently don't), this allows the expected results to be specified without the need to explicitly construct a `BinaryInstruction` in those cases, relying instead on an implicit conversion.